### PR TITLE
fix(rust): correct AExpr.to_field for bitwise and logical and/or

### LINKED ISSUE #16359: `expr.to_field` returns wrong datatype for bitwise and/or
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example

```rust
use polars::prelude::*;
use polars_plan::logical_plan::Context;

fn main() -> Result<(), PolarsError> {
    let df = df!("a" => &[1], "b" => &[3])?.lazy();
    let schema = &df.schema()?;
    let expr = col("a").and(col("b"));
    let out = df.select([expr.clone()]);
    println!("{}", &out.collect()?);
    let dtype = expr.to_field(schema, Context::Default)?;
    println!("{:?}", dtype);
    Ok(())
}
```

### Log output

```shell
run ProjectionExec
```

### Issue description

`col.and` and `col.or` are described as being the bitwise and and or respectively. However, `to_field` treats them as logical and/or instead, returning the wrong (boolean) datatype when asked.

### Expected behavior

Bitwise operations should preserve datatype in the logical plan query side (the correct output is determined by the physical executor).

### Installed versions

<details>

polars = { version = "0.39.2", features = ["lazy"] }

</details>

### PR DESCRIPTION
**Pull request by wence-**
AExpr.to_field would incorrectly state that the bitwise and/or of integral columns was boolean, and conversely that the logical and/or was the common supertype. Fix this by matching on the correct operator names.

- Closes #16359

### CODE REVIEW

**Comment by wence-** [comment]
TODO: Where is the right place to put tests for this?

**Comment by ritchie46** [comment]
> TODO: Where is the right place to put tests for this?

On the python side. 

Something like this

```python
pl.LazyFrame(..).select(my_expr).schema == ...
```

File: py-polars/src/lazyframe/visitor/expr_nodes.rs [hunk 1]
```diff
@@ -42,6 +42,7 @@ pub enum PyClosedInterval {
     Both,
     Left,
     Right,
+    #[pyo3(name="None_")]
     None,
```
Comment (wence-) [inline py-polars/src/lazyframe/visitor/expr_nodes.rs:46]: This fixes a naming issue I introduced in #16355, can spin out if you like.

Comment (wence-) [reply py-polars/src/lazyframe/visitor/expr_nodes.rs]: Removed here.

**Review by ritchie46 — APPROVED** [review]

**Pull request merged**